xdsl/dialects/experimental/math.py

```diff
@@ -95,6 +93,20 @@ class Atan2Op(IRDLOperation):
     rhs: Operand = operand_def(AnyFloat)
     result: OpResult = result_def(AnyFloat)
 
+    # TODO: use __init__
+    def __init__(
+        self,
+        lhs: Operation | SSAValue,
+        rhs: Operation | SSAValue,
+        fastmath: FastMathFlagsAttr | None = None,
+    ):
+        lhs = SSAValue.get(lhs)
```

Comment:
this isn't necessary `IRDLOperation.__init__` accepts `Operation | SSAValue`

Reply:
Currently I'm doing     
```
return super().__init__(
            attributes=attributes,
            operands=[lhs, rhs],
            result_types=[SSAValue.get(lhs).type],
        )
```
Which is what's currently been used in gpu dialect. I'm not sure whether this is the best way.